src/main/java/org/craftercms/studio/impl/v2/service/policy/validators/PathPolicyValidator.java

```diff
@@ -58,18 +60,27 @@ private void validatePermitted(HierarchicalConfiguration<?> config, Action actio
         var sourceRegex = config.getString(CONFIG_KEY_SOURCE_REGEX);
         if (!item.matches(sourceRegex)) {
             String modifiedItem = null;
+            StringBuilder reasonBuilder = new StringBuilder();
             var targetRegex = config.getString(CONFIG_KEY_TARGET_REGEX);
             if (targetRegex != null) {
                 modifiedItem = item.replaceAll(sourceRegex, targetRegex);
-
+                if (!modifiedItem.equalsIgnoreCase(item)) {
+                    addPolicyReason(reasonBuilder, format("Path is replaced from source regex '%s' to target regex '%s'.", sourceRegex, targetRegex));
+                }
                 var caseTransform = config.getString(CONFIG_KEY_CASE_TRANSFORM);
                 if (isNotEmpty(caseTransform)) {
                     switch (caseTransform.toLowerCase()) {
                         case "uppercase":
-                            modifiedItem = modifiedItem.toUpperCase();
+                            if (modifiedItem.chars().anyMatch(c -> Character.isLowerCase(c))) {
+                                addPolicyReason(reasonBuilder, "Path is transformed to uppercase.");
+                                modifiedItem = modifiedItem.toUpperCase();
+                            }
                             break;
                         case "lowercase":
-                            modifiedItem = modifiedItem.toLowerCase();
+                            if (modifiedItem.chars().anyMatch(c -> Character.isUpperCase(c))) {
+                                addPolicyReason(reasonBuilder, "Path is transformed to lowercase.");
```

Comment:
Maybe we use the one reason as I mention above.

Reply:
I updated the reason message.